app/models/article.rb

```diff
@@ -2,31 +2,41 @@
 #
 # Table name: articles
 #
-#  id                       :bigint           not null, primary key
-#  content                  :text
-#  description              :text
-#  status                   :integer
-#  title                    :string
-#  views                    :integer
-#  created_at               :datetime         not null
-#  updated_at               :datetime         not null
-#  account_id               :integer          not null
-#  author_id                :bigint
-#  category_id              :integer
-#  folder_id                :integer
-#  portal_id                :integer          not null
+#  id                    :bigint           not null, primary key
+#  content               :text
+#  description           :text
+#  status                :integer
+#  title                 :string
+#  views                 :integer
+#  created_at            :datetime         not null
+#  updated_at            :datetime         not null
+#  account_id            :integer          not null
+#  associated_article_id :bigint
+#  author_id             :bigint
+#  category_id           :integer
+#  folder_id             :integer
+#  portal_id             :integer          not null
 #
 # Indexes
 #
-#  index_articles_on_author_id                 (author_id)
+#  index_articles_on_associated_article_id  (associated_article_id)
+#  index_articles_on_author_id              (author_id)
 #
 # Foreign Keys
 #
+#  fk_rails_...  (associated_article_id => articles.id)
 #  fk_rails_...  (author_id => users.id)
 #
 class Article < ApplicationRecord
   include PgSearch::Model
 
+  has_many :associated_articles,
+           class_name: :Article,
+           foreign_key: :associated_article_id,
+           dependent: :nullify,
+           inverse_of: 'associated_article'
+
+  belongs_to :associated_article, class_name: :Article, optional: true
```

Comment:
little confused here since `linked_article` along with `linked_articles` could cause confusion. But let's sync up and ensure a consistent naming across categories and articles.

Reply:
I couldn't think of a better inverse association for linked_articles.